### PR TITLE
Refactor(eos_cli_config_gen): Remove EOS default configuration

### LINKED ISSUE #4329: (eos_cli_config_gen) always generate hostname
### Enhancement summary

Hi, can we please change the `eos_cli_config_gen/templates/eos/hostnamej2` template  so it always generates the hostname? Or is it required by the CVP to remove it? Right now it can be disabled with the `generate_default_config` flag set to `false`. We use the flag to skip lines like `no enable password` and `no aaa root` which are truly default and don't show up in the running config. But the hostname always appear in the running config.

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

We use the results of `eos_cli_config_gen` as a tool to generate the indented config which later we use as a basis for the running config compliance checks. It is important to all the non-default lines to be present in the config and all the default config lines to be removed. Most of the lines get correctly removed with `eos_cli_config_gen` - the only exception is the `hostname` config that we think should be always present.

### Describe the solution you would like

Remove the conditional:
```
{% if not generate_default_config is arista.avd.defined(false) %}
!
hostname {{ hostname | arista.avd.default(inventory_hostname) }}
{% endif %}
```

### Describe alternatives you have considered

Unconditionally print the hostname
```
!
hostname {{ hostname | arista.avd.default(inventory_hostname) }}
```

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Remove EOS default configuration

## Component(s) name

`arista.avd.eos_cli_config_gen`

### Default EOS configuration is no longer automatically generated

The `generate_default_config` setting controls generation of some default EOS configuration. This was previously set to `true` by default.

As of AVD 5.0.0 the default value of `generate_default_config` has been changed to `false` to prevent generation of any configuration not specified by the
inputs to `eos_cli_config_gen` (structured configuration).

The following commands will be generated when `generate_default_config` is set to `true`:

- RANCID Content Type
- Hostname
- Default configuration `no aaa root`
- Default configuration `no enable password`
- Transceiver qsfp default mode
- End of configuration delimiter

The setting `generate_default_config` will be removed in AVD 6.0.0.

The following `eos_cli_config_gen` inputs can be given as needed to generate the same configuration:

```diff
-generate_default_config: true
+aaa_root:
+  disabled: true
+config_comment: "RANCID-CONTENT-TYPE: arista"
+config_end: true
+enable_password:
+  disabled: true
+hostname: "{{ inventory_hostname }}"
+transceiver_qsfp_default_mode_4x10: true
```

All changes except `"RANCID-CONTENT-TYPE: arista"` have been incorporated into `eos_designs`. Action is only required when using `eos_cli_config_gen` directly,
or if the `RANCID-CONTENT-TYPE: arista` is required.

## How to test

Updated existing molecule for the `false` case to instead test the `true` case.
All other molecule inputs left untouched, so the changes reflects what users will see.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/aaa-root.j2 [hunk 1]
```diff
@@ -4,10 +4,10 @@
...
-{% if aaa_root is arista.avd.defined %}
-{%     if aaa_root.secret.sha512_password is arista.avd.defined %}
+{% if aaa_root.disabled is arista.avd.defined(true) %}
+no aaa root
+{% elif aaa_root.secret.sha512_password is arista.avd.defined %}
 aaa root secret sha512 {{ aaa_root.secret.sha512_password | arista.avd.hide_passwords(hide_passwords) }}
-{%     endif %}
-{% elif not generate_default_config is arista.avd.defined(false) %}
+{% elif generate_default_config is arista.avd.defined(true) %}
 no aaa root
```
Comment (MaheshGSLAB) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/aaa-root.j2:12]:
```suggestion
{% if aaa_root.disabled is arista.avd.defined(true) or generate_default_config is arista.avd.defined(true) %}
no aaa root
{% elif aaa_root.secret.sha512_password is arista.avd.defined %}
aaa root secret sha512 {{ aaa_root.secret.sha512_password | arista.avd.hide_passwords(hide_passwords) }}
```

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/enable-password.j2 [hunk 2]
```diff
@@ -5,15 +5,18 @@
...
 {%         if enable_password.hash_algorithm is arista.avd.defined('md5') %}
 enable password 5 {{ enable_password.key | arista.avd.hide_passwords(hide_passwords) }}
 {%         elif enable_password.hash_algorithm is arista.avd.defined('sha512') %}
 enable password sha512 {{ enable_password.key | arista.avd.hide_passwords(hide_passwords) }}
 {%         endif %}
 {%     endif %}
-{% elif not generate_default_config is arista.avd.defined(false) %}
+{% elif generate_default_config is arista.avd.defined(true) %}
 !
 no enable password
```
Comment (MaheshGSLAB) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/enable-password.j2:21]:
```suggestion
{%     if enable_password.disabled is arista.avd.defined(true) or generate_default_config is arista.avd.defined(true) %}
!
no enable password
{%     elif enable_password.key is arista.avd.defined %}
!
{%         if enable_password.hash_algorithm is arista.avd.defined('md5') %}
enable password 5 {{ enable_password.key | arista.avd.hide_passwords(hide_passwords) }}
{%         elif enable_password.hash_algorithm is arista.avd.defined('sha512') %}
enable password sha512 {{ enable_password.key | arista.avd.hide_passwords(hide_passwords) }}
{%         endif %}
{%     endif %}
```

File: ansible_collections/arista/avd/molecule/eos_designs-l2ls/documentation/devices/BGP-LEAF1.md [hunk 3]
```diff
@@ -53,6 +55,12 @@ management api http-commands
       no shutdown
 ```
 
+## Authentication
+
+### Enable Password
+
+Enable password has been disabled
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/molecule/eos_designs-l2ls/documentation/devices/BGP-LEAF1.md:62]: same as enabled password do we need to add aaa root authentication has been disabled?

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/role-configuration.md [hunk 4]
```diff
@@ -79,15 +79,17 @@ roles/eos_cli_config_gen/docs/tables/eos-cli-config-gen-documentation.md
 
 ### Generate default config
 
-The `generate_default_config` knob allows to omit default EOS configuration.
-This can be useful when leveraging `eos_cli_config_gen` to generate configlets with CloudVision.
+The `generate_default_config` setting controls generation of some default EOS configuration.
 
-The following commands will be omitted when `generate_default_config` is set to `false`:
+This is historic behavior which will be removed in AVD 6.0.0.
+Instead it is recommended to build structured config according to the intended output configurations.
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/role-configuration.md]:
```suggestion
The `generate_default_config` setting controls the generation of some default EOS configurations.

This is historic behavior which will be removed in AVD 6.0.0.
Instead it is recommended to build structured config according to the intended output configurations.
```

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 5]
```diff
@@ -268,6 +278,41 @@ As of AVD 5.0.0 the automatic conversion from both of the previous models has be
               route_map: static-to-bgp
     ```
 
+### Default EOS configuration is no longer automatically generated
+
+The `generate_default_config` setting controls generation of some default EOS configuration. This was previously set to `true` by default.
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]:
```suggestion
The `generate_default_config` setting controls the generation of some default EOS configurations. This was previously set to `true` by default.
```

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/aaa-root.j2 (hunk 1 above)
Comment (ClausHolbechArista) [reply python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/aaa-root.j2:12]: No the order here is important to retain historic behavior.

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/enable_password.schema.yml [hunk 6]
```diff
@@ -9,6 +9,10 @@ keys:
   enable_password:
     type: dict
     keys:
+      disabled:
+        type: bool
```
Comment (laxmikantchintakindi) [inline python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/enable_password.schema.yml:13]:
```suggestion
        type: bool
        required: true
```

File: ansible_collections/arista/avd/molecule/eos_designs-l2ls/documentation/devices/BGP-LEAF1.md (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_designs-l2ls/documentation/devices/BGP-LEAF1.md:62]: we do not have any docs for aaa root today, so no. We were already disabling aaa root all the time.

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/enable_password.schema.yml (hunk 6 above)
Comment (ClausHolbechArista) [reply python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/enable_password.schema.yml:13]: This is not required. If someone sets hash_algorithm. This is different than the places where we have `enabled` to trigger a feature to be configured at all.

**Review by MaheshGSLAB — APPROVED** [review]

**Review by laxmikantchintakindi — APPROVED** [review]

**Pull request merged**